backend/app/tests/test_services/test_memory_regression.py

```
@@ -0,0 +1,120 @@
+class TestMemoryRegression:
+    """Guard against dependency-graph memory leaks."""
+
+    # ── Configurable thresholds ─────────────────────────────────────
+    # RSS is measured inside pytest which already loaded the full app,
+    # so absolute values include all imports + conftest fixtures.
+    # The growth test is the meaningful regression detector.
+    RSS_GROWTH_MAX = 15.0  # MB — request handling should not balloon
+
+    def test_rss_growth_after_requests(self, _app_and_client):
+        """RSS should not grow excessively after handling 200+ requests."""
+        app, client = _app_and_client
```

Ruff (0.15.21)

[warning] Unpacked variable `app` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)